neural_speed/convert/convert_chatglm.py

```diff
@@ -973,7 +1115,9 @@ def main(args_in: Optional[List[str]] = None) -> None:
     # ChatGLM3 shares the same architecture and model config with ChatGLM2
     # but its tokenizer further supports system prompts,
     # so we can check system token to discriminate ChatGLM3 from ChatGLM2.
-    if hasattr(tokenizer, "tokenizer") and "<|system|>" in tokenizer.tokenizer.special_tokens:
+    if hasattr(model.config, "rope_ratio"):
```

Comment:
please add some annotations here to explain why using this `if-statement` to choose related converting functions.